internal/provider/resource_cloudspace/cloudspace_resource_gen.go

```diff
@@ -75,16 +75,17 @@ func CloudspaceResourceSchema(ctx context.Context) schema.Schema {
 			"deployment_type": schema.StringAttribute{
 				Optional:            true,
 				Computed:            true,
-				Description:         "The name of the cloudspace.",
-				MarkdownDescription: "The name of the cloudspace.",
+				Description:         "Specifies the deployment type for the cloudspace (Only gen2 is allowed value).",
+				MarkdownDescription: "Specifies the deployment type for the cloudspace (Only gen2 is allowed value).",
+				DeprecationMessage:  "Support will be limited to Gen2 cloudspaces, rendering this field obsolete.",
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.RequiresReplace(),
 					stringplanmodifier.UseStateForUnknown(),
 				},
 				Validators: []validator.String{
-					stringvalidator.RegexMatches(regexp.MustCompile(`^gen[12]$`), "deployment_type must be gen1 (or) gen2"),
+					stringvalidator.OneOf("gen1", "gen2"),
```

Comment:
We only want to support `gen2` right? Why are we still supporting `gen1` as a valid input here?

Reply:
Corrected